### ISSUE TITLE
Apollo适配PostgreSQL数据库

### ISSUE THREAD

**Issue by liutao5121** [apolloconfig/apollo#4532]
apollo使用版本：1.5.1
postgreSQL驱动：
`  <dependency>
            <groupId>org.postgresql</groupId>
            <artifactId>postgresql</artifactId>
            <version>42.4.0</version>
        </dependency>`

适配改动主要点如下：
1、修改初始化sql脚本（ApolloConfigDB.sql、ApolloPortalDB.sql），
a、修改3个字段的类型，如下：
IsDeleted smallint DEFAULT 0 ,
IsAbandoned boolean DEFAULT false NOT NULL,
IsPublic boolean DEFAULT false NOT NULL,
b、pg索引是全局的，注意索引唯一性（重复的可以使用，下划线+数字 解决此问题）
c、sql语句中表名和字段名全部不加双引号、反引号
d、去掉语句中的(191)，直接替换为空

2、application-github.propperties中的配置如下：
spring.jpa.database-platform = org.hibernate.dialect.PostgreSQLDialect
spring.datasource.driver-class-name = org.postgresql.Driver
spring.jpa.properties.hibernate.globally_quoted_identifiers=false
spring.jpa.hibernate.globally_quoted_identifiers=false
spring.jpa.properties.hibernate.temp.use_jdbc_metadata_defaults=false
spring.datasource.hikari.connectionInitSql=

3、启动脚本中增加环境变量配置：
-Duser.timezone=Asia/Chongqing

4、修改BaseEntity类里的isDelete的类型为int
protected int isDeleted = 0;
  public int isDeleted() {
    return isDeleted;
  }
public void setDeleted(boolean deleted) {
    isDeleted = (deleted == true ? 1 : 0);
  }

**Comment by nobodyiam** [comment]
Thanks for the information! 

> 1、修改初始化sql脚本（ApolloConfigDB.sql、ApolloPortalDB.sql），
a、修改3个字段的类型，如下：
IsDeleted smallint DEFAULT 0 ,
IsAbandoned boolean DEFAULT false NOT NULL,
IsPublic boolean DEFAULT false NOT NULL,
b、pg索引是全局的，注意索引唯一性（重复的可以使用，下划线+数字 解决此问题）
c、sql语句中表名和字段名全部不加双引号、反引号
d、去掉语句中的(191)，直接替换为空

Is this change compatible with MySQL?

**Comment by liutao5121** [comment]
no, it's not mysql compatible, It's just for adapting to pgsql.Generally speaking,one application corresponds to one database.
We only need to execute the corresponding initialization sql script.

**Comment by pankajbsn** [comment]
Apollo ticked all my boxes, except would love to have a postgres version.

**Comment by nobodyiam** [comment]
Thanks for the feedback. PostgreSQL is still not an officially supported server database in the current Apollo distribution. The repository packages the PostgreSQL JDBC driver and has a `postgre` profile placeholder, but it does not provide PostgreSQL initialization/upgrade scripts or an official PostgreSQL deployment path; the deployment guide still points to older community adaptations. Therefore, we cannot recommend the Apollo 1.5.1 changes in this issue as a supported setup for current releases.

A maintained contribution would need current `ApolloConfigDB` and `ApolloPortalDB` initialization/migration scripts, compatibility across Config Service, Admin Service, and Portal, and automated PostgreSQL integration tests. Contributions or a concrete design proposal covering those pieces would be welcome.

Note: this reply was generated and posted automatically by AI for initial triage; a maintainer will follow up if needed.
